src/xunit.analyzers/AssertIsTypeShouldNotBeUsedForAbstractTypeFixer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Xunit.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class AssertIsTypeShouldNotBeUsedForAbstractTypeFixer : CodeFixProvider
+    {
+        private const string Title = "Use Assert." + IsAssignableFrom;
+
+        private const string IsAssignableFrom = "IsAssignableFrom";
+
+        private const string IsType = "IsType";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Descriptors.X2017_AssertIsTypeShouldNotBeUsedForAbstractType.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var invocation = root.FindNode(context.Span).FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            var methodName = invocation.GetGenericName()?.Identifier.Text;
+            if (methodName == IsType)

```

Comment:
Way I see it, there are 2 ways a user might end up with the analyzer firing for a `IsNotType`:
1. They had just written the assert and we know 100% that it will always succeed, thus it's unnecessary (although could serve documentation purposes, I suppose?)
2. They are refactoring an existing codebase with existing tests and modifying the class hierarchy. As a result a number of existing valid asserts become effective no-ops. Having a fixer that can be applied in batch mode would allow them get the repo back to no warnings. Of course the appropriate fix might be to change the type that's asserted, but having the option might be nice.